src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -392,4 +402,238 @@ public static MetadataServiceClient getMetadataServiceClient(GoogleCredentials c
+      }
+    }
+
+    Schema.Type avroType = schema.getType();
+    switch (avroType) {
+      case RECORD:
+        return Type.RECORD;
+      case STRING:
+        return Type.STRING;
+      case FLOAT:
+        return Type.FLOAT;
+      case DOUBLE:
+        return Type.DOUBLE;
+      case BOOLEAN:
+        return Type.BOOLEAN;
+      case NULL:
+        return Type.NULL;
+      case BYTES: // BYTES is binary data with variable size.
+        return Type.BINARY;
+      case INT:
+        return Type.INT32;
+      case LONG:
+        return Type.INT64;
+      case UNION: // Shouldn't happen. Unions can not contain other unions as per Avro spec.
+      case ARRAY: // Not supported as a primitive type (e.g. if this is an ARRAY of ARRAYs).
+      case MAP:
+      case ENUM:
+      default:
+        return Type.TYPE_UNSPECIFIED;
+    }
+  }
+
+  private static Type dataplexLogicalFieldType(Schema schema) {
+    LogicalType logicalType = schema.getLogicalType();
+
+    if (logicalType == LogicalType.DECIMAL) {
+      return Type.DECIMAL;
+    } else if (logicalType == LogicalType.DATE) {
+      return Type.DATE;
+    } else if (logicalType == LogicalType.TIME_MICROS
+      || logicalType == LogicalType.TIME_MILLIS) {
+      return Type.TIME;
+    } else if (logicalType == LogicalType.TIMESTAMP_MICROS
+      || logicalType == LogicalType.TIMESTAMP_MILLIS) {
+      return Type.TIMESTAMP;
+    }
+
+    return null;
+  }
+
+  /**
+   * Add partition info to dataplex entity.
+   *
+   * @param entity dataplex entity
+   * @param credentials Google Credentials
+   * @param bucketName
+   * @param tableName
+   * @param project
+   * @throws IOException
+   */
+  public static void addPartitionInfo(Entity entity,
+                                      GoogleCredentials credentials,
+                                      String bucketName, String tableName, String project) throws IOException {
+    Storage storage = GCPUtils.getStorage(project, credentials);
+    String delimiter = "/";
+    String partitionPrefix = DataplexConstants.STORAGE_BUCKET_PARTITION_KEY + "=";
+    Page<Blob> blobs =
+      storage.list(
+        bucketName,
+        Storage.BlobListOption.prefix(tableName + delimiter + partitionPrefix),
+        Storage.BlobListOption.currentDirectory());
+    String lastPartition = null;
+    for (Blob blob : blobs.iterateAll()) {
+      lastPartition = blob.getName();
+    }
+    String location = "gs://" + bucketName +
+      delimiter + lastPartition.substring(0, lastPartition.length() - 1);
+    String[] lastPartitionParts = lastPartition.split(delimiter);
+    Partition.Builder dataplexPartitionBuilder =
+      Partition.newBuilder();
+    try (MetadataServiceClient metadataServiceClient =
+           getMetadataServiceClient(credentials)) {
+      Partition partition = dataplexPartitionBuilder
+        .setLocation(location)
+        .addValues(lastPartitionParts[lastPartitionParts.length - 1].substring(partitionPrefix.length()))
```

Comment:
some comments to explain what we are doing and why would help here.
- Why do we need to get the last blob name
- show example of what does the blob name look like, to make the parsing logic easier to understand